src/btree/bt_split.c

```diff
@@ -1526,9 +1526,11 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
             /*
              * If we have written a prepared update, we need to retain the next update that is not a
              * tombstone. Otherwise, we don't have anything to write in the next reconciliation if
-             * the prepared update is reverted.
+             * the prepared update is reverted. If the next value update is a modify, we need to
+             * retain all the older updates until a full value is found.
              */
             if (WT_TIME_WINDOW_HAS_START_PREPARE(&supd->tw)) {
+                bool find_next_value = false;
```

Comment:
I didn't clearly understand the use of this `find_next_value` flag. Doesn't adding `if (WT_UPDATE_DATA_VALUE(tmp))` code before `break` solve the requirement?

Reply:
This is to exclude the tombstone. We need a full standard update. I can simplify this to:

if (tmp->type == WT_UPDATE_STANDARD)
        break;

Reply:
Yes. that would be good.